src/main/java/org/eclipse/tractusx/ssi/lib/jwt/SignedJwtFactory.java

```diff
@@ -50,57 +48,138 @@ public class SignedJwtFactory {
-   * all private key types are possible, in the context of Distributed Identity using an Elliptic
-   * Curve key ({@code P-256}) is advisable.
+   * Creates a signed JWT {@link SignedJWT} that contains a set of claims and an issuer with a
+   * specific configuration {@link JwtConfig} for VP
    *
-   * @param didIssuer the did issuer
-   * @param audience the value of the token audience claim, e.g. the IDS Webhook address.
-   * @param serializedPresentation the serialized presentation
-   * @param privateKey the private key
-   * @return a {@code SignedJWT} that is signed with the private key and contains all claims listed.
+   * @param didIssuer
+   * @param audience
+   * @param serializedPresentation
+   * @param privateKey
+   * @param keyId
+   * @param config
+   * @return
    */
   @SneakyThrows
   public SignedJWT create(
-      URI id,
       Did didIssuer,
       String audience,
       SerializedVerifiablePresentation serializedPresentation,
-      IPrivateKey privateKey) {
+      IPrivateKey privateKey,
+      String keyId,
+      JwtConfig config) {
 
     final String issuer = didIssuer.toString();
     final String subject = didIssuer.toString();
 
+    TypeReference<HashMap<String, Object>> typeRef =
+        new TypeReference<HashMap<String, Object>>() {};
+
     // make on object out of it so that it can get serialized again
     Map<String, Object> vp =
-        new ObjectMapper().readValue(serializedPresentation.getJson(), HashMap.class);
+        new ObjectMapper().readValue(serializedPresentation.getJson(), typeRef);
+
+    Date iat = new Date();
 
     var claimsSet =
         new JWTClaimsSet.Builder()
             .issuer(issuer)
             .subject(subject)
             .audience(audience)
             .claim("vp", vp)
-            .expirationTime(new Date(new Date().getTime() + 60 * 1000))
-            .jwtID(id.toString())
+            .issueTime(iat)
+            .expirationTime(new Date(iat.getTime() + config.getExpirationTime() * 1000))
+            .jwtID(UUID.randomUUID().toString())
             .build();
 
     final OctetKeyPair octetKeyPair = octetKeyPairFactory.fromPrivateKey(privateKey);
-    return createSignedES256Jwt(octetKeyPair, claimsSet, issuer);
+    return createSignedES256Jwt(octetKeyPair, claimsSet, issuer, keyId);
   }
 
-  private static SignedJWT createSignedES256Jwt(
-      OctetKeyPair privateKey, JWTClaimsSet claimsSet, String issuer) {
+  /**
+   * Creates a signed JWT {@link SignedJWT} from a Verifiable Credential
+   *
+   * @param didIssuer
+   * @param holderIssuer
+   * @param vc
+   * @param privateKey
+   * @param keyId
+   * @return
+   */
+  @SneakyThrows
+  public SignedJWT create(
+      Did didIssuer,
+      Did holderIssuer,
+      LinkedHashMap<String, Object> vc,
+      IPrivateKey privateKey,
+      String keyId) {
+    final String issuer = didIssuer.toString();
+    final String subject = holderIssuer.toString();
```

Comment:
Why are these two equal?

Reply:
its not equal one is holderDid and one is Issuer did for iss and sub in the jwt https://www.w3.org/TR/vc-data-model/ probably the naming was confusing

Reply:
I guess I misread that. All good